ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/hierarchies/OntologyHierarchiesPresenterImpl.java

```diff
@@ -0,0 +1,333 @@
+import com.sencha.gxt.data.shared.Store;
+import com.sencha.gxt.data.shared.TreeStore;
+import com.sencha.gxt.widget.core.client.TabItemConfig;
+import com.sencha.gxt.widget.core.client.TabPanel;
+import com.sencha.gxt.widget.core.client.tree.Tree;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author aramsey
+ */
+public class OntologyHierarchiesPresenterImpl implements OntologyHierarchiesView.Presenter,
+                                                         OntologyHierarchySelectionChangedEvent.OntologyHierarchySelectionChangedEventHandler,
+                                                         AppRatingSelected.AppRatingSelectedHandler,
+                                                         AppRatingDeselected.AppRatingDeselectedHandler,
+                                                         AppFavoriteSelectedEvent.AppFavoriteSelectedEventHandler {
+
+
+    class AppAVUCallback implements AsyncCallback<List<Avu>> {
+
+        AppDetailsDialog dlg;
+        App app;
+        List<OntologyHierarchy> hierarchies;
+        List<List<String>> appGroupHierarchies;
+
+        public AppAVUCallback(AppDetailsDialog dlg, App app) {
+            this.dlg = dlg;
+            this.app = app;
+        }
+        @Override
+        public void onFailure(Throwable caught) {
+            ErrorHandler.post(caught);
+        }
+
+        @Override
+        public void onSuccess(List<Avu> result) {
+            // Create list of group hierarchies
+            hierarchies = convertAvusToHierarches(result);
+            appGroupHierarchies = ontologyUtil.getAllPathsList(hierarchies);
+
+            dlg.show(app,
+                     searchRegexPattern,
+                     appGroupHierarchies,
+                     OntologyHierarchiesPresenterImpl.this,
+                     OntologyHierarchiesPresenterImpl.this,
+                     OntologyHierarchiesPresenterImpl.this);
+        }
+
+        List<OntologyHierarchy> convertAvusToHierarches(List<Avu> avus) {
+            List<OntologyHierarchy> selectedHierarchies = Lists.newArrayList();
+            for (Avu avu: avus) {
+                List<OntologyHierarchy> hierarchies = iriToHierarchyMap.get(avu.getValue());
+                if (hierarchies != null) {
+                    selectedHierarchies.addAll(hierarchies);
+                }
+            }
+            return selectedHierarchies;
+        }
+    }
+    class AppDetailsCallback implements AsyncCallback<App> {
+
+        private final AppDetailsDialog dlg;
+
+        public AppDetailsCallback(AppDetailsDialog dlg) {
+            this.dlg = dlg;
+        }
+
+        @Override
+        public void onFailure(final Throwable caught) {
+            announcer.schedule(new ErrorAnnouncementConfig(appearance.fetchAppDetailsError(caught)));
+        }
+
+        @Override
+        public void onSuccess(final App result) {
+            serviceFacade.getAppAVUs(result, new AppAVUCallback(dlg, result));
```

Comment:
Not for this PR, but it might be better if the details endpoint returns the app's metadata, rather than requiring the UI to chain these calls.